File: API/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace API.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow;
        public int CustomerId { get; set; }
        public AppUser Customer { get; set; }
        public string FullName { get; set; }
        public string Destination { get; set; }
        public string PhoneNumber { get; set; }
        public ICollection<OrderDetail> OrderDetails { get; set; }
        public int? PromotionId { get; set; }
        public Promotion Promotion { get; set; }
        public int ShippingMethodId { get; set; }
        public ShippingMethod ShippingMethod { get; set; }
        public int PaymentMethodId { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public double SubTotal { get; set; }
        public double Discount { get; set; }
        public int State { get; set; }
    }
}

File: client/src/app/_models/order.ts
export interface OrderDetail {
    productCode: string;
    productName: string;
    photoUrl: string;
    colorCode: string;
    colorName: string;
    price: string;
    quantity: number;
}

export interface Promotion {
    id: number;
    promotionCode: string;
    discountPercent: number;
    maxDiscountAmount: number;
    minOrderTotal: number;
    start: Date;
    end: Date;
}

export interface ShippingMethod {
    id: number;
    name: string;
    minTime: number;
    maxTime: number;
    price: number;
    state: boolean;
}

export interface PaymentMethod {
    id: number;
    name: string;
    state: boolean;
}

export interface Order {
    id: number;
    date: Date;
    fullName: string;
    destination: string;
    phoneNumber: string;
    orderDetails: OrderDetail[];
    promotion: Promotion;
    shippingMethod: ShippingMethod;
    paymentMethod: PaymentMethod;
    subTotal: number;
    discount: number;
    state: number;
}